### PR TITLE
Fixes for Jammy CI

### PR DESCRIPTION
**Pull request by bluca**

### CODE REVIEW

**Comment by bluca** [comment]
Interesting, there's a build failure on jammy ppc64el that shows up nowhere else:

```
../src/shared/efi-api.c: In function ‘efi_get_reboot_to_firmware’:
../src/shared/efi-api.c:158:16: error: ‘b’ may be used uninitialized [-Werror=maybe-uninitialized]
  158 |         return !!(b & EFI_OS_INDICATIONS_BOOT_TO_FW_UI);
      |                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from ../src/basic/macro.h:12,
                 from ../src/basic/alloc-util.h:9,
                 from ../src/shared/efi-api.c:5:
../src/shared/efi-api.c: In function ‘efi_set_reboot_to_firmware’:
../src/fundamental/macro-fundamental.h:329:34: error: ‘b’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  329 |         ((b) ? ((orig) | (flag)) : ((orig) & ~(flag)))
      |                                  ^
../src/shared/efi-api.c:163:18: note: ‘b’ was declared here
  163 |         uint64_t b, b_new;
```

**Comment by yuwata** [comment]
Please add a brief comment about the warning. Otherwise, we may optimize the code in the future.

**Comment by bluca** [comment]
TEST-13-NSPAWN on jammy-ppc64el:

```
Apr 30 00:38:17 H testsuite-13.sh[1241]: Applying NETLINK_AUDIT mask on architecture: ppc
Apr 30 00:38:17 H testsuite-13.sh[1241]: Failed to add audit seccomp rule, ignoring: Invalid argument
Apr 30 00:38:17 H testsuite-13.sh[1241]: Applying NETLINK_AUDIT mask on architecture: ppc64
Apr 30 00:38:17 H testsuite-13.sh[1241]: Failed to add audit seccomp rule, ignoring: Invalid argument
Apr 30 00:38:17 H testsuite-13.sh[1241]: Applying NETLINK_AUDIT mask on architecture: ppc64-le
Apr 30 00:38:17 H testsuite-13.sh[1241]: Failed to add audit seccomp rule, ignoring: Invalid argument
Apr 30 00:38:17 H testsuite-13.sh[1241]: Dropping capabilities failed: Invalid argument
Apr 30 00:38:17 H testsuite-13.sh[1239]: Child died too early.
```

**Comment by yuwata** [comment]
Are there any progress on this?

> TEST-13-NSPAWN on jammy-ppc64el:
> 
> ```
> Apr 30 00:38:17 H testsuite-13.sh[1241]: Applying NETLINK_AUDIT mask on architecture: ppc
> Apr 30 00:38:17 H testsuite-13.sh[1241]: Failed to add audit seccomp rule, ignoring: Invalid argument
> Apr 30 00:38:17 H testsuite-13.sh[1241]: Applying NETLINK_AUDIT mask on architecture: ppc64
> Apr 30 00:38:17 H testsuite-13.sh[1241]: Failed to add audit seccomp rule, ignoring: Invalid argument
> Apr 30 00:38:17 H testsuite-13.sh[1241]: Applying NETLINK_AUDIT mask on architecture: ppc64-le
> Apr 30 00:38:17 H testsuite-13.sh[1241]: Failed to add audit seccomp rule, ignoring: Invalid argument
> Apr 30 00:38:17 H testsuite-13.sh[1241]: Dropping capabilities failed: Invalid argument
> Apr 30 00:38:17 H testsuite-13.sh[1239]: Child died too early.
> ```

I cannot find the relevant logs?

**Comment by bluca** [comment]
```
Oct 02 20:45:01 H systemd[1]: Finished System Power Off.
Oct 02 20:45:01 H systemd[1]: Forcibly powering off: unit systemd-poweroff.service succeeded
Oct 02 20:45:01 H systemd[1]: poweroff.target changed dead -> active
Oct 02 20:45:01 H systemd[1]: poweroff.target: Job 396 poweroff.target/start finished, result=done
Oct 02 20:45:01 H systemd[1]: Reached target System Power Off.
Oct 02 20:45:01 H audit: BPF prog-id=0 op=UNLOAD
Oct 02 20:45:01 H audit: BPF prog-id=0 op=UNLOAD
Oct 02 20:45:01 H systemd[1]: Shutting down.
Oct 02 20:45:01 H systemd[1]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=500 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Oct 02 20:45:01 H systemd[1]: Assertion '*_head == _item' failed at src/core/device.c:69, function device_unset_sysfs(). Aborting.
Oct 02 20:45:01 H systemd[1]: Caught <ABRT> from our own process.
```

**Comment by bluca** [comment]
```
Oct 02 21:28:43 H systemd[1]: Reached target System Power Off.
Oct 02 21:28:43 H systemd[1]: Shutting down.
Oct 02 21:28:43 H systemd[1]: systemd-journald.service: Releasing resources.
Oct 02 21:28:43 H systemd[1]: systemd-journald.service: Releasing all stored fds
Oct 02 21:28:43 H systemd[1]: Sent message type=method_call sender=n/a destination=org.freedesktop.DBus path=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RemoveMatch cookie=4254 reply_cookie=0 signature=s error-name=n/a error-message=n/a
Oct 02 21:28:43 H systemd[1]: Bus bus-api-system: changing state RUNNING → CLOSED
Oct 02 21:28:43 H systemd[1]: Assertion 'hashmap_isempty(m->units_requiring_mounts_for)' failed at src/core/manager.c:1559, function manager_free(). Aborting.
Oct 02 21:28:43 H systemd[1]: Caught <ABRT> from our own process.
```

**Comment by bluca** [comment]
Have you seen those 2 asserts before @mrc0mmand ? They happened on Jammy/ppc64el

**Comment by mrc0mmand** [comment]
> Have you seen those 2 asserts before @mrc0mmand ? They happened on Jammy/ppc64el

Interesting, I've never seen them before, I wonder why they popped up only on Jammy.

File: test/units/testsuite-70.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ env PIN=123456 /usr/lib/systemd/systemd-cryptsetup attach test-volume $img - tpm
 env PIN=123457 /usr/lib/systemd/systemd-cryptsetup attach test-volume $img - tpm2-device=auto,headless=1 && { echo 'unexpected success'; exit 1; }
 
 # Check LUKS2 token plugin unlock (i.e. without specifying tpm2-device=auto)
-if cryptsetup --help | grep -q 'LUKS2 external token plugin support is compiled-in'; then
+if cryptsetup --help | grep -q 'LUKS2 external token plugin support is compiled-in' && [ -f "$(cryptsetup --help | grep -F 'LUKS2 external token plugin path' | sed 's|LUKS2 external token plugin path: \(.*\)\.|\1|g')/libcryptsetup-token-systemd-tpm2.so" ]; then
```
Comment (keszybz) [inline test/units/testsuite-70.sh]:
```
[ -f "$(cryptsetup --help | sed -n -r 's/.*LUKS2 external token plugin path: (.*)\./\1/p')/libcryptsetup-token-systemd-tpm2.so" ]
```

Maybe insert a line break somewhere?

**Pull request merged**